x-pack/plugins/apm/common/utils/to_kuery_filter_format.ts

```diff
@@ -5,10 +5,34 @@
  * 2.0.
  */
 
+import { isEmpty } from 'lodash';
+
+type Separator = 'OR' | 'AND';
+
 export function toKueryFilterFormat(
   key: string,
   values: string[],
-  separator: 'OR' | 'AND' = 'OR'
+  separator: Separator = 'OR'
 ) {
   return values.map((value) => `${key} : "${value}"`).join(` ${separator} `);
 }
+export function mergeKueries(
+  kuery1: string,
+  kuery2: string,
```

Comment:
nit: These names are a bit confusing. By looking at the usages, could it be something like:

`kuery1` = `hostNameKuery`
`kuery2` = `filterKuery`

Reply:
this function is not specific to hostnames or Profiling filters. It should be generic in a way that anyone can use to join multiple kueries. 

I liked @mykolaharmash's idea to make the args an array, this will also improve the naming.